### PR TITLE
Added config for param indentation.

### PR DESCRIPTION
**Pull request by commondream**
Currently vim-ruby aligns parameter lists containing line breaks with the first parameter in the parameter list, but our team's style guide calls for aligning using a standard indent on the next line instead, so I end up cleaning up lines any time I have a line break in a method call or definition. This PR adds the g:ruby_indent_param_style option with two possible values, "param" or "line", to make indentation within parameter lists configurable.

### CODE REVIEW

**Comment by AndrewRadev** [comment]
The way that vim-ruby works right now, there are two valid styles of indenting arguments:

``` ruby
some_method(arg1,
            arg2)

some_method(
  arg1,
  arg2
)
```

It's *really* difficult to support all possible indentation styles and combinations. We would *really* like not to. If you could use the second style in your style guide, that would be great.

Still, you've made an option, I guess it's no big deal, I can only hope that this doesn't cause trouble down the line. I'm willing to merge this. But it's kind of important whether you've covered all your bases. For instance, even with your patch, this indents like so:

``` ruby
some_method({arg1: arg2,
             arg2: arge})
```

Is this something that makes sense in your style guide? Do you only align arguments like this, or do you do the same for hashes? What about other "hanging" styles of indentation? For instance:

``` ruby
@foo ||= begin
           foo
         end
```

Basically, could you describe your style guide in a bit more detail?

**Comment by commondream** [comment]
We use https://github.com/bbatsov/ruby-style-guide, the style guide written up by the Rubucop guy. There are a few optional items in the style guide, so like when we insert a line break in a parameter list, we indent the next line an additional two spaces. You're right on the hash - we would have only indented it two spaces. I'll try to find a way to pull that in as well. This style, though, is one that I've seen in a lot of code, so I don't think it's out there to include it. Please let me know, though, if you'd rather not, and we can avoid any extra work for either of us.

**Comment by AndrewRadev** [comment]
Okay, first off, I'm not a big fan of that guide in general, but let's say that's a personal preference.

Secondly, the guide explicitly says:

``` ruby
# starting point (line is too long)
def send_mail(source)
  Mailer.deliver(to: 'bob@example.com', from: 'us@example.com', subject: 'Important message', body: source.text)
end

# bad (double indent)
def send_mail(source)
  Mailer.deliver(
      to: 'bob@example.com',
      from: 'us@example.com',
      subject: 'Important message',
      body: source.text)
end

# good
def send_mail(source)
  Mailer.deliver(to: 'bob@example.com',
                 from: 'us@example.com',
                 subject: 'Important message',
                 body: source.text)
end

# good (normal indent)
def send_mail(source)
  Mailer.deliver(
    to: 'bob@example.com',
    from: 'us@example.com',
    subject: 'Important message',
    body: source.text
  )
end
```

Both styles that are supported by vim-ruby are listed as "good". Your described style doesn't seem to be there at all.

For the record, though, indentation works like that with no brackets:

``` ruby
some_method arg1,
  arg2
```

> This style, though, is one that I've seen in a lot of code, so I don't think it's out there to include it.

Fair enough, but as I said, I'd like to at least see a consistent set of "this is how we want things to end up". In the end, this is something we'll have to support, so it would be good to have an idea of "this is the style we're looking for", "this is the reason we have this option". We've had *a lot* of edge cases in the past, and we need a bird-eye view of how things are supposed to work.

For instance, "indent arguments two shiftwidths, but option hashes only one shiftwidth" is probably not something we'll include. It sounds way too specific for no apparent reason, and it's hard for me to imagine it being a popular style. The style guide you linked to has no mention of it, unless I'm missing some "optional" tag or a discussion in the issues that describes it.

> Please let me know, though, if you'd rather not, and we can avoid any extra work for either of us.

I can't promise you I'll merge this, but if you tell me it's a reasonably popular style, I'll consider it. Even if I don't, you could use a company fork for it, so I don't think it's wasted effort on your part. Vim-ruby doesn't update that often, and I think you'll be able to keep up with changes.

**Comment by commondream** [comment]
I just talked to our team, and they're cool with taking on the new line after paren version, so I'll just close this out.
